lang/cs/Org.Apache.REEF.Common/metrics/MutableMetricsLayer/MutableCounter.cs

```diff
@@ -0,0 +1,114 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System;
+using System.Reactive;
+using Org.Apache.REEF.Common.metrics.Api;
+using Org.Apache.REEF.Utilities.Logging;
+
+namespace Org.Apache.REEF.Common.Metrics.MutableMetricsLayer
+{
+    /// <summary>
+    /// Mutable Counter metric class.
+    /// </summary>
+    public sealed class MutableCounter : MutableMetricBase
+    {
+        private long _value = 0;
+        private readonly object _lock = new object();
+        private static readonly Logger Logger = Logger.GetLogger(typeof(MutableCounter));
+
+        /// <summary>
+        /// Counter Constructor
+        /// </summary>
+        /// <param name="info">Meta-data info. of the metric.</param>
+        /// <param name="initValue">Initial counter value.</param>
+        public MutableCounter(IMetricsInfo info, long initValue)
```

Comment:
Why is this `public`? Wouldn't an instance of this always be returned by methods in `DefaultMetricsSourceImpl`?